src/cargo/ops/cargo_install.rs

```diff
@@ -39,6 +40,26 @@ impl Drop for Transaction {
     }
 }
 
+enum RustupToolchainSource {
```

Comment:
As far as I know, rustup hasn't yet claimed those values are stable. At least the doc doesn't say anything more than the environment key <https://github.com/rust-lang/rustup/blob/de2b979bd4569cacfabf501bb58464c9139b0d6d/doc/user-guide/src/environment-variables.md?plain=1#L75>.

* Could we have a link to the rustup source code if we want to make an assumption on that?
* Should we check with the rustup team to ensure these value strings are stable?

Reply:
I am happy to get confirmation from the Rustup team that the five values are stable, but would you feel more comfortable if they were listed directly in the documentation?